src/main/java/fi/helsinki/cs/tmc/core/commands/factory/CommandFactory.java

```diff
@@ -0,0 +1,223 @@
+package fi.helsinki.cs.tmc.core.commands.factory;
+
+import fi.helsinki.cs.tmc.core.cache.ExerciseChecksumCache;
+import fi.helsinki.cs.tmc.core.commands.*;
+import fi.helsinki.cs.tmc.core.communication.*;
+import fi.helsinki.cs.tmc.core.communication.updates.ExerciseUpdateHandler;
+import fi.helsinki.cs.tmc.core.communication.updates.ReviewHandler;
+import fi.helsinki.cs.tmc.core.configuration.TmcSettings;
+import fi.helsinki.cs.tmc.core.domain.Course;
+import fi.helsinki.cs.tmc.core.domain.Exercise;
+import fi.helsinki.cs.tmc.core.domain.ProgressObserver;
+import fi.helsinki.cs.tmc.core.domain.Review;
+import fi.helsinki.cs.tmc.core.domain.submission.SubmissionResult;
+import fi.helsinki.cs.tmc.core.exceptions.TmcCoreException;
+import fi.helsinki.cs.tmc.core.spyware.DiffSender;
+import fi.helsinki.cs.tmc.core.zipping.ProjectRootFinder;
+import fi.helsinki.cs.tmc.langs.abstraction.ValidationResult;
+import fi.helsinki.cs.tmc.langs.domain.RunResult;
+import fi.helsinki.cs.tmc.langs.util.TaskExecutorImpl;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Handles the creation of different Commands. Mainly used by
+ * {@link fi.helsinki.cs.tmc.core.TmcCore}
+ */
+
+public class CommandFactory {
+
+    /**
+     * .
```

Comment:
Don't really like how these JavaDocs are -- we should also avoid just having at-clauses w/o real description